### PR TITLE
Implement `isNot`, `notIn`, `hasNot`  comparators.

### PR DESCRIPTION
**Pull request by BorysShulyak**
type: **minor**

## Changes
- `isProtectedEntity`
  - Refactor `isProtectedEntity` method using **Strategy** pattern.
  - Implement `isNot`, `notIn`, `hasNot`  comparators.

### CODE REVIEW

**Comment by BorysShulyak** [comment]
Close as duplicate of https://github.com/mattmilburn/strapi-plugin-do-not-delete/pull/8

**Comment by BorysShulyak** [comment]
@mattmilburn Could you review the PR, please?

File: server/utils/is-protected-entity.js [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+};
+
+const COMPARATOR_ACTION_STRATEGY = {
+  [COMPARATORS.is]: (value, entityAttr) => value === entityAttr,
+  [COMPARATORS.isNot]: (value, entityAttr) => value !== entityAttr,
+  [COMPARATORS.in]: (value, entityAttr) => value.includes(entityAttr),
+  [COMPARATORS.notIn]: (value, entityAttr) => !value.includes(entityAttr),
+  [COMPARATORS.has]: (value, entityAttr) => entityAttr.includes(value),
+  [COMPARATORS.hasNot]: (value, entityAttr) => !entityAttr.includes(value),
+  [COMPARATORS.matches]: (value) => RegExp(value).test(),
```
Comment (BorysShulyak) [inline server/utils/is-protected-entity.js]:
```suggestion
  [COMPARATORS.matches]: (value, entityAttr) => RegExp(value).test(entityAttr),
```

**Comment by BorysShulyak** [comment]
@mattmilburn Covered with tests. Could you review the PR, please? Thank you!

**Comment by mattmilburn** [comment]
@BorysShulyak Tests are looking good! Did you see my other 2 comments in `is-protected-entity.js`?

**Comment by BorysShulyak** [comment]
> @BorysShulyak Tests are looking good! Did you see my other 2 comments in `is-protected-entity.js`?

@mattmilburn Unfortunately no, I don't see any other comments from your side on this merge request (

**Comment by mattmilburn** [comment]
@BorysShulyak Ah, strange. Okay I re-posted those 2 comments. Let me know if you still can't see them.

**Comment by BorysShulyak** [comment]
> @BorysShulyak Ah, strange. Okay I re-posted those 2 comments. Let me know if you still can't see them.

- https://gyazo.com/b1e7b66c2212c74d4e899874a06c696b 
- https://gyazo.com/840998b01ea5d55bc89a82cbb76d1c86
@mattmilburn I actually still don't see them ((

**Comment by BorysShulyak** [comment]
Only 7 conversations in the PR https://gyazo.com/067f0a6eaccaaa66cdae6dfe554a6869

**Comment by mattmilburn** [comment]
@BorysShulyak Can you see them at these links?

* https://github.com/mattmilburn/strapi-plugin-do-not-delete/pull/8/files#diff-b7fe4d8527b350aa4687da08bdf6dfea704ec5a9c5887d9366f0309e73e06de6R11
* https://github.com/mattmilburn/strapi-plugin-do-not-delete/pull/8/files#diff-b7fe4d8527b350aa4687da08bdf6dfea704ec5a9c5887d9366f0309e73e06de6R14

**Comment by BorysShulyak** [comment]
@mattmilburn Nothing (
https://gyazo.com/40a448cd456d008e2188a821e3d98057
I have also tried logging out and checking on the mobile - no result.

**Comment by mattmilburn** [comment]
@BorysShulyak Haha I've never had this happen before. Okay I'll post them here 🙂 
Screenshot 2023-07-21 at 10 12 47 AM
Screenshot 2023-07-21 at 10 12 56 AM

**Comment by BorysShulyak** [comment]
@mattmilburn Thank you! I have applied your suggestions.
For the first point - it was really an overengineering. And renamed the argument for the second point.

**Comment by mattmilburn** [comment]
@BorysShulyak Excellent work 🎉  Merging into staging now 👍🏻

**Pull request merged**